src/reconcile/rec_visibility.c

```diff
@@ -436,6 +418,25 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
         }
     }
 
+    /*
+     * If we found a tombstone with a time pair earlier than the update it applies to, which can
+     * happen if the application performs operations with timestamps out-of-order, make it invisible
+     * by making the start time pair match the stop time pair of the tombstone. We don't guarantee
+     * that older readers will be able to continue reading content that has been made invisible by
+     * out-of-order updates.
+     */
+    if (upd_select->stop_ts < upd_select->start_ts ||
+      (upd_select->stop_ts == upd_select->start_ts &&
+          upd_select->stop_txn < upd_select->start_txn)) {
```

Comment:
We get out of order transaction ids?

Reply:
I think we can with lower isolation levels (e.g., updates performed with `isolation=read-committed`).

Reply:
I should clarify that I'm not suggesting that lost updates are a feature, but if the API allows it I don't think we should assert either.